extensions/wikia/FeedsAndPosts/ArticleData.class.php

```diff
@@ -5,19 +5,26 @@
 use ArticleService;
 use ImageServing;
 use Title;
+use WikiaDataAccess;
 
 class ArticleData {
-	public static function getImages(int $articleId, $limit=11): array {
-		$imageData = ( new ImageServing( [ $articleId ] ) )->getImages( $limit )[strval( $articleId )] ?? [];
+	public static function getImages( int $articleId, $limit = 11 ): array {
+		return WikiaDataAccess::cache(
+			wfMemcKey('feeds', 'article-images'),
+			10800, // 3h
+			function () use ( $articleId, $limit ) {
+				$imageData = ( new ImageServing( [ $articleId ] ) )->getImages( $limit )[strval( $articleId )] ?? [];
```

Comment:
Let's split this into multiple lines, it's a bit difficult to follow.

Reply:
I tried various ways and all of them did not look better.. Actually this single line version seems to be most readable

Reply:
I was thinking of something like
```
$imageServing = new ImageServing( [ $articleId ] );
$images = $imageServing->getImages( $limit );

$imageInfo = $images[strval( $articleId )] ?? [];

Reply:
k, makes sense